qutebrowser/browser/webkit/network/pac.py

```diff
@@ -0,0 +1,529 @@
+# vim: ft=python fileencoding=utf-8 sts=4 sw=4 et:
+
+# Copyright 2014-2016 Florian Bruhin (The Compiler) <mail@qutebrowser.org>
+#
+# This file is part of qutebrowser.
+#
+# qutebrowser is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# qutebrowser is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with qutebrowser.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Evaluation of PAC scripts."""
+
+import sys
+import ipaddress
+import fnmatch
+import traceback
+import itertools
+import functools
+from PyQt5.QtCore import QObject, QVariant, QMetaType, QDateTime, pyqtSlot
+from PyQt5.QtQml import QJSEngine, QJSValue
+from PyQt5.QtNetwork import QHostInfo, QNetworkProxy
+
+
+class ParseProxyError(Exception):
+    """Error in parsing PAC result string."""
+
+    pass
+
+
+class EvalProxyError(Exception):
+    """Error in evaluating PAC script."""
+
+    pass
+
+
+class _PACContext(QObject):
+    """Implementation of PAC API.
+
+    See http://web.archive.org/web/20060424005037/wp.netscape.com/eng/mozilla/2.0/relnotes/demo/proxy-live.html
+    """
+
+    #pylint: disable=no-method-argument
+    def _js_slot(*args):
```

Comment:
Notice that this uses internal class field (namely, `self._error` -- is it still okay to move this outside of a class?